src/chrome/internal/sources/resourceIdentifier.ts

```diff
@@ -0,0 +1,222 @@
+        return true;
+    }
+
+    protected abstract generateCanonicalized(): string;
+
+    public toString(): string {
+        return `res:${this.textRepresentation}`;
+    }
+
+    constructor(textRepresentation: TString) {
+        super(textRepresentation);
+        this._canonicalized = this.generateCanonicalized();
+    }
+}
+
+// A unix local resource location is a *nix path
+// e.g.: /home/user/proj/myfile.js
+export class UnixLocalFilePath<TString extends string = string> extends LocalFilePathCommonLogic<TString> implements ILocalFilePath<TString> {
+    protected generateCanonicalized(): string {
+        const normalized = path.normalize(this.textRepresentation); // Remove ../s
+        return normalized.replace(/(?:\\\/|\/)+/, '/');
+    }
+
+    public static isValid(path: string) {
+        return path.startsWith('/');
+    }
+}
+
+// A windows local file path
+// e.g.: C:\proj\myfile.js
+export class WindowLocalFilePath<TString extends string = string> extends LocalFilePathCommonLogic<TString> implements ILocalFilePath<TString> {
+    protected generateCanonicalized(): string {
+        const normalized = path.normalize(this.textRepresentation); // Remove ../s
+        return normalized.toLowerCase().replace(/[\\\/]+/, '\\');
+    }
+
+    public static isValid(path: string) {
+        return path.match(/^[A-Za-z]:/);
+    }
+
+    public get canonicalized(): string {
+        return this.textRepresentation.toLowerCase().replace(/[\\\/]+/, '\\');
+    }
+}
+
+// Any file path that we don't recognize as Windows nor Linux
+export class UnrecognizedFilePath<TString extends string = string> extends IsEquivalentAndConstructorCommonLogic<TString> implements ILocalFilePath<TString> { }
+
+function parseWindowsOrUnixLocalResourcePath<TString extends string = string>(path: TString): ILocalFilePath<TString> | null {
+    if (WindowLocalFilePath.isValid(path)) {
+        return new WindowLocalFilePath<TString>(path);
+    } else if (UnixLocalFilePath.isValid(path)) {
+        return new UnixLocalFilePath<TString>(path);
+    } else {
+        return null;
+    }
+}
+
+function parseLocalResourcePath<TString extends string = string>(path: TString): ILocalFilePath<TString> {
+    const recognizedLocalResourcePath = parseWindowsOrUnixLocalResourcePath<TString>(path);
+    if (recognizedLocalResourcePath !== null) {
+        return recognizedLocalResourcePath;
+    } else {
+        return new UnrecognizedFilePath<TString>(path);
+    }
+}
+
+function parseURL<TString extends string = string>(textRepresentation: TString): URL<TString> {
+    if (LocalFileURL.isValid(textRepresentation)) {
+        return new LocalFileURL<TString>(textRepresentation);
+    } else {
+        return new NonLocalFileURL<TString>(textRepresentation);
+    }
+}
+
+/**
+ * Sample formats:
+ * file:///D:\\scripts\\code.js
+ * file:///Users/me/project/code.js
+ * c:/scripts/code.js
+ * http://site.com/scripts/code.js
+ * http://site.com/
```

Comment:
Would be great to see unit tests covering these cases.

Reply:
We are definitely going to add unit tests for these cases. We are planning on adding them after V2 stabilizes, and we don't have any remaining big changes to do to it.